Library/Homebrew/cask/cask.rb

```diff
@@ -440,6 +440,8 @@ def to_hash_with_variations(hash_method: :to_h)
           MacOSVersion::SYMBOLS.keys.product(OnSystem::ARCH_OPTIONS).each do |os, arch|
             bottle_tag = ::Utils::Bottles::Tag.new(system: os, arch:)
             next unless bottle_tag.valid_combination?
+            next if depends_on.macos && !bottle_tag.to_macos_version.compare(depends_on.macos.comparator,
```

Comment:
Not sure what happened to it, but I distinctly remember @dduugg adding a method similar to `#compare` before. If that's still around we should use that. Maybe @dduugg will remember the details.

Reply:
I believe `compare` was the method I added 😹: https://github.com/Homebrew/brew/pull/15152/files

Reply:
Since `MacOSVersion` inherits from `Version`, the one I added handles arrays of `MacOSVersion`s and defers to `super` otherwise. Not sure if having all the functionality in the original method is preferred.

Reply:
Oops, missed that this was a specialisation. I thought the original method was defined only for `MacOSVersion` 😅